spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/dsl/ComposedTaskParser.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.core.dsl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static org.springframework.cloud.dataflow.core.dsl.TokenKind.*;
+
+/**
+ * Parse a composed task specification.
+ *
+ * @author Andy Clement
+ */
+public class ComposedTaskParser {
+
+	private Tokens tokens;
+
+	public ComposedTaskParser() {
+	}
+
+	/**
+	 * Parse a composed task definition into an abstract syntax tree (AST).
+	 *
+	 * @param composedTaskSpecification the textual composed task specification
+	 * @param validate if true then extra validation will be performed on parsed AST
+	 * @return the AST for the parsed composed task
+	 * @throws ComposedTaskSpecificationException if any problems occur during parsing
+	 */
+	public ComposedTaskNode parse(String composedTaskSpecification) {
```

Comment:
Need to add the taskName to the params.  The TaskParser currently validates that the taskName is valid.